src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -165,155 +292,847 @@ private MessageConsumer newReplyStream(
+                .put(updatedFrameHeader.buffer(), updatedFrameHeader.offset(), updatedFrameHeader.sizeof())
+                .put(attach.buffer(), attach.offset(), attach.sizeof())
                 .build();
-            receiver.accept(data.typeId(), data.buffer(), data.offset(), data.sizeof());
+
+            doNetworkData(traceId, authorization, 0L, attachWithFrameHeader);
         }
 
-        private void doAmqpOpen(
-            AmqpFrameFW frame)
+        private void doEncodeTransfer(
+            long traceId,
+            long authorization,
+            int channel,
+            long handle,
+            int flags,
+            OctetsFW extension,
+            OctetsFW payload,
+            int payloadIndex)
         {
-            // TODO
-            /*
-            final AmqpOpenFrameFW open = amqpOpenFrameRW.wrap(encodeBuffer, 0, encodeBuffer.capacity())
-                .length(frame.length())
-                .dof(frame.dof())
-                .type(frame.type())
-                .channel(frame.channel())
-                .performative(frame.performative())
-                .payload(frame.payload)
+            final AmqpDataExFW dataEx = extension.get(amqpDataExRO::tryWrap);
+            byte[] deliveryTagBytes = new byte[1];
+            dataEx.deliveryTag().bytes().get((b, o, l) -> deliveryTagBytes[0] = b.getByte(o));
+            BoundedOctets16FW deliveryTag = boundedOctetsRW.wrap(boundedOctetsBuffer, 0, boundedOctetsBuffer.capacity())
+                .set(deliveryTagBytes)
+                .build();
+            int flag = dataEx.flags();
+            int bitmask = 1;
+            int settled = 0;
+            if ((flag & bitmask) == bitmask)
+            {
+                settled = 1;
+                bitmask = bitmask << 2;
+            }
+            // TODO: add cases for resume, aborted, batchable
+
+            AmqpFrameHeaderFW frameHeader =
+                amqpFrameHeaderRW.wrap(frameBuffer, 0, frameBuffer.capacity())
+                    .size(0)
+                    .doff(2)
+                    .type(0)
+                    .channel(channel)
+                    .performative(b -> b.domainId(d -> d.set(DESCRIBED))
+                                        .descriptorId(d2 -> d2.set(AmqpFrameType.TRANSFER.value())))
+                    .build();
+
+            int payloadSize = payload.sizeof();
+
+            final int frameHeaderSize = frameHeader.sizeof();
+            final int transferOffset = frameHeader.limit();
+            AmqpTransferFW.Builder transferBuilder =  amqpTransferRW.wrap(frameBuffer, transferOffset, frameBuffer.capacity());
+            AmqpTransferFW transfer = payloadIndex > 0 ? transferBuilder.handle(handle).build() :
+                transferBuilder.handle(handle)
+                    .deliveryId(dataEx.deliveryId())
+                    .deliveryTag(deliveryTag)
```

Comment:
Seems like you want a `.deliveryTag(OctetsFW)` setter as well, even though `deliveryTag` is a `BoundedOctetsFW`.